inductiva/structures/scenarios/plate_linear_elastic/geometry_utils.py

```diff
@@ -0,0 +1,427 @@
+        super().__init__("circular", center_x, center_y)
+        self.radius = radius
+
+
+class RectangularHole(Hole):
+    """Rectangular hole.
+
+    Attributes:
+        center_x (float): x-coordinate of the center.
+        center_y (float): y-coordinate of the center.
+        size_x (float): Size of the hole in the x-direction.
+        size_y (float): Size of the hole in the y-direction.
+        angle (float): Positive angle of rotation in degrees around the hole 
+          center.
+    """
+
+    def __init__(self, center_x: float, center_y: float, size_x: float,
+                 size_y: float, angle: float) -> None:
+        """Initializes a RectangularHole object."""
+        super().__init__("rectangular", center_x, center_y)
+        self.size_x = size_x
+        self.size_y = size_y
+        self.angle = angle
+
+
+class EllipticalHole(Hole):
+    """Elliptical hole.
+
+    Attributes:
+        center_x (float): x-coordinate of the center.
+        center_y (float): y-coordinate of the center.
+        radius_x (float): Radius of the hole in the x-direction.
+        radius_y (float): Radius of the hole in the y-direction.
+        angle (float): Positive angle of rotation in degrees around the hole
+          center."""
+
+    def __init__(self, center_x: float, center_y: float, radius_x: float,
+                 radius_y: float, angle: float) -> None:
+        """Initializes a EllipticalHole object."""
+        super().__init__("elliptical", center_x, center_y)
+        self.radius_x = radius_x
+        self.radius_y = radius_y
+        self.angle = angle
+
+
+class GeometricCase:
+    """Geometric case.
+
+    The geometric case is characterized by a plate and a set of holes.
+
+    Attributes:
+        plate_object (RectangularPlate): Plate object.
+        list_holes_object (List[Hole]): The holes objects.
+    """
+
+    def __init__(self,
+                 plate_object: RectangularPlate,
+                 list_holes_objects: Optional[List[Hole]] = None) -> None:
+        """Initializes a GeometricCase object."""
+        self.plate_object = plate_object
+        self.list_holes_objects = list_holes_objects
+
+    def write_to_json(self, json_path: str) -> None:
+        """Writes the geometric case to JSON file.
+
+        Args:
+            json_path (str): The JSON file path.
+        """
+
+        # Plate dictionary
+        plate_dict = {"plate": vars(self.plate_object)}
+
+        # Holes dictionary
+        if self.list_holes_objects is not None:
+
+            all_hole_properties = []
+            for hole_object in self.list_holes_objects:
+                hole_dictionary = vars(hole_object)
+                all_hole_properties.append(hole_dictionary)
+
+            holes_dict = {"holes": all_hole_properties}
+
+            # Merge dictionaries: geometric case dictionary
+            geom_case_dictionary = {**plate_dict, **holes_dict}
+        else:
+            geom_case_dictionary = {**plate_dict}
```

Comment:
You can avoid this if-else block. Just populate the list of holes with `[]` if you get `None` for it on `__init__`. Then the dictionary of the geometry case always has a "holes" key, that may or not be empty.